src/compiler/checker.ts

```diff
@@ -15958,7 +15958,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function createSignatureTypeMapper(signature: Signature, typeArguments: readonly Type[] | undefined): TypeMapper {
-        return createTypeMapper(signature.typeParameters!, typeArguments);
+        return createTypeMapper(map(signature.typeParameters!, tp => tp.mapper ? instantiateType(tp, tp.mapper) : tp), typeArguments);

```

Comment:
The problem is that the `signature` might already be a product of `instantiateSignature` that was called without `eraseTypeParameters` and which thus cloned type parameters. Those cloned parameters are instantiated there in the arguments list using the available `mapper` but the same mapper wasn't later used when erasing type parameters from that result. This led to the mapper-instantiated cloned type parameter being left non-instantiated after erasing type parameters.

Reply:
Not 100% certain, but it may be better to do `sameMap` here just because that would avoid extra arrays when nothing actually needs to be mapped (but maybe it doesn't happen that often).

Reply:
sure, I pushed out a commit to use `sameMap` here